allennlp/models/dependency_parser.py

```diff
@@ -0,0 +1,464 @@
+        """
+        Parameters
+        ----------
+        tokens : Dict[str, torch.LongTensor], required
+            The output of ``TextField.as_array()``, which should typically be passed directly to a
+            ``TextFieldEmbedder``. This output is a dictionary mapping keys to ``TokenIndexer``
+            tensors.  At its most basic, using a ``SingleIdTokenIndexer`` this is: ``{"tokens":
+            Tensor(batch_size, num_tokens)}``. This dictionary will have the same keys as were used
+            for the ``TokenIndexers`` when you created the ``TextField`` representing your
+            sequence.  The dictionary is designed to be passed directly to a ``TextFieldEmbedder``,
+            which knows how to combine different word representations into a single vector per
+            token in your input.
+        tags : torch.LongTensor, optional (default = None)
+            A torch tensor representing the sequence of integer gold class labels of shape
+            ``(batch_size, num_tokens)``.
+
+        Returns
+        -------
+        An output dictionary consisting of:
+        loss : torch.FloatTensor, optional
+            A scalar loss to be optimised.
+        """
+        embedded_text_input = self.text_field_embedder(words)
+        if pos_tags is not None and self._pos_tag_embedding is not None:
+            embedded_pos_tags = self._pos_tag_embedding(pos_tags)
+            embedded_text_input = torch.cat([embedded_text_input, embedded_pos_tags], -1)
+        elif self._pos_tag_embedding is not None:
+            raise ConfigurationError("Model uses a POS embedding, but no POS tags were passed.")
+
+        mask = get_text_field_mask(words)
+        float_mask = mask.float()
+        encoded_text = self.encoder(embedded_text_input, mask)
+
+        # shape (batch_size, timesteps, arc_representation_dim)
+        head_arc_representation = F.elu(self.head_arc_projection(encoded_text))
+        child_arc_representation = F.elu(self.child_arc_projection(encoded_text))
+
+        # shape (batch_size, timesteps, type_representation_dim)
+        head_type_representation = F.elu(self.head_type_projection(encoded_text))
+        child_type_representation = F.elu(self.child_type_projection(encoded_text))
+        head_type_representation = head_type_representation.contiguous()
+        child_type_representation = child_type_representation.contiguous()
+        # shape (batch_size, timesteps, timesteps)
+        attended_arcs = self.arc_attention(head_arc_representation,
+                                           child_arc_representation,
+                                           mask.float(),
+                                           mask.float()).squeeze(1)
+
+        minus_inf = -1e8
+        minus_mask = (1 - float_mask) * minus_inf
+        attended_arcs = attended_arcs + minus_mask.unsqueeze(2) + minus_mask.unsqueeze(1)
+
+        if self.training or (self.eval and not self.use_mst_decoding_for_validation):
```

Comment:
isn't `self.eval` a function, so it will always evaluate as truthy? possibly you just want

```
if self.training or not self.use_mst_decoding_for_validation:
```

Reply:
Good catch, thanks 👍